Lib/contextlib.py

```diff
@@ -152,13 +164,20 @@ def __exit__(self, typ, value, traceback):
                 try:
                     raise RuntimeError("generator didn't stop")
                 finally:
+                    del self.exc_context
```

Comment:
How about set it back to `None`?

```suggestion
                    self.exc_context = None
```

or, better:

```py
    def __exit__(self, typ, value, traceback):
        exc_context = self.exc_context
        self.exc_context = None
```